docs/reference-guides/block-api/block-bindings.md

```diff
@@ -0,0 +1,296 @@
+
+Block Bindings API lets you “bind” dynamic data to the block’s attributes, which are then reflected in the final HTML markup that is output to the browser on the front end.
+
+An example could be connecting an Image block `url` attribute to a function that returns random images from an external API.
+
+```html
+<!-- wp:image {
+	"metadata":{
+		"bindings":{
+			"url":{
+				"source":"my-plugin/get-random-images"
+			}
+		}
+	}
+} -->
+```
+
+
+## Compatible blocks and their attributes
+
+Right now, not all block attributes are compatible with block bindings. There is some effort going on increasing this compatibility, but for now, this is the list:
+
+- Paragraph block:
+    - content
+
+- Heading block:
+    - content
+
+- Image block:
+    - ID
+    - url
+    - title
+    - alt
+
+- Button block:
+    - text
+    - url
+    - linkTarget
+    - rel
+
+## Registering a custom source
+
+Registering a source requires defining at least `name` and a `callback` function that gets a value from the source and passes it back to a block attribute.
+
+Once a source is registered, any block that supports the Block Bindings API can use a value from that source by setting its `metadata.bindings` attribute to a value that refers to the source.
+
+Registration can be done on the server via PHP or in the editor via JavaScript, and both can coexist.
+
+The label defined in server registration will be overridden by the label defined in the editor.
+
+### Server registration
+
+Server registration allows applying a callback that will be executed on the frontend for the defined bound attribute.
+
+The function to register a custom source is `register_block_bindings_source($name, $args)`.
+
+- `name` is a `string` that sets the unique ID for the custom source.
+- `args` is an `array` that contains:
+    - `label` is a `string` with the human-readable name of the custom source.
+    - `uses_context` is an `array` with the block context that is passed to the callback (optional).
+    - `get_value_callback` is the `function` that will run on the bound block render function. It accepts three arguments: `source_args`, `block_instance` and `attribute_name`. This value can be overriden with the filter `block_bindings_source_value`.
+
+Note that `register_block_bindings_source()` should be called from a handler attached to the `init` hook.
+
+Here is an example:
+
+```php
+add_action(
+	'init',
+	function() {
+		register_block_bindings_source(
+			'wpmovies/visualization-date',
+			array(
+				'label'              => __( 'Visualization Date', 'custom-bindings' ),
+				'get_value_callback' => function( array $source_args, $block_instance ) {
+					$post_id = $block_instance->context['postId'];
+					$visualization_date = get_post_meta( $post_id, 'wp_movies_visualization_date', true );
+					if ( ! $visualization_date ) {
+						$visualization_date = date("m/d/Y");
+					}
+					return $visualization_date;
+				},
+				'uses_context'       => array( 'postId', 'postType' ),
```

Comment:
Maybe this is a minor thing, but I don't see the post type used. Do we need `postType` here?

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/66251/commits/f578dbcf95fdcbda2c0cb1f8b2e2238506e1e13e